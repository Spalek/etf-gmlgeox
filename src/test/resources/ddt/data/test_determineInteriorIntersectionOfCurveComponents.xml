<ii:GeometryCollection xmlns:gml="http://www.opengis.net/gml/3.2"
 xmlns:ii="http://www.interactive-instruments.de/test" xmlns:xlink="http://www.w3.org/1999/xlink"
 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
 <gml:boundedBy>
  <gml:Envelope srsName="http://www.opengis.net/def/crs/EPSG/0/4326">
   <gml:lowerCorner>0 0</gml:lowerCorner>
   <gml:upperCorner>90 90</gml:upperCorner>
  </gml:Envelope>
 </gml:boundedBy>
 <ii:member>
  <gml:Curve gml:id="c1">
   <gml:segments>
    <gml:LineStringSegment>
     <gml:posList>1 1 3 1</gml:posList>
    </gml:LineStringSegment>
   </gml:segments>
  </gml:Curve>
 </ii:member>
 <ii:member>
  <gml:Curve gml:id="c2">
   <gml:segments>
    <gml:LineStringSegment>
     <gml:posList>1 2 3 2</gml:posList>
    </gml:LineStringSegment>
   </gml:segments>
  </gml:Curve>
 </ii:member>
 <ii:member>
  <gml:Curve gml:id="c3">
   <gml:segments>
    <gml:LineStringSegment>
     <gml:posList>2 1 2 3</gml:posList>
    </gml:LineStringSegment>
   </gml:segments>
  </gml:Curve>
 </ii:member>
 <ii:member>
  <gml:Curve gml:id="c4">
   <gml:segments>
    <gml:LineStringSegment>
     <gml:posList>4 1 5 1</gml:posList>
    </gml:LineStringSegment>
   </gml:segments>
  </gml:Curve>
 </ii:member>
 <ii:member>
  <gml:Point gml:id="p1">
   <gml:pos>4.5 1</gml:pos>
  </gml:Point>
 </ii:member>
 <ii:member>
  <gml:Point gml:id="p2">
   <gml:pos>5 1</gml:pos>
  </gml:Point>
 </ii:member>
 <ii:member>
  <gml:Curve gml:id="c5">
   <gml:segments>
    <gml:LineStringSegment>
     <gml:posList>4 2 5 2</gml:posList>
    </gml:LineStringSegment>
   </gml:segments>
  </gml:Curve>
 </ii:member>
 <ii:member>
  <gml:Curve gml:id="c6">
   <gml:segments>
    <gml:LineStringSegment>
     <gml:posList>5 2 4 2</gml:posList>
    </gml:LineStringSegment>
   </gml:segments>
  </gml:Curve>
 </ii:member>
 <ii:member>
  <gml:Curve gml:id="c7">
   <gml:segments>
    <gml:LineStringSegment>
     <gml:posList>7 1 8 1</gml:posList>
    </gml:LineStringSegment>
    <gml:Arc>
     <gml:posList>8 1 8.5 1.5 9 1</gml:posList>
    </gml:Arc>
    <gml:LineStringSegment>
     <gml:posList>9 1 10 1</gml:posList>
    </gml:LineStringSegment>
   </gml:segments>
  </gml:Curve>
 </ii:member>
 <ii:member>
  <gml:Curve gml:id="c8">
   <gml:segments>
    <gml:LineStringSegment>
     <gml:posList>10 1 9 1</gml:posList>
    </gml:LineStringSegment>
    <gml:Arc>
     <gml:posList>9 1 8.5 1.5 8 1</gml:posList>
    </gml:Arc>
    <gml:LineStringSegment>
     <gml:posList>8 1 7 1</gml:posList>
    </gml:LineStringSegment>
   </gml:segments>
  </gml:Curve>
 </ii:member>
 <ii:member>
  <gml:Curve gml:id="c9">
   <gml:segments>
    <gml:LineStringSegment>
     <gml:posList>7 3 9 3</gml:posList>
    </gml:LineStringSegment>    
   </gml:segments>
  </gml:Curve>
 </ii:member>
 <ii:member>
  <gml:CompositeCurve gml:id="c10">
   <gml:curveMember>
    <gml:Curve gml:id="c10_1">
     <gml:segments>
      <gml:LineStringSegment>
       <gml:posList>7 3 8 3</gml:posList>
      </gml:LineStringSegment>      
     </gml:segments>
    </gml:Curve>
   </gml:curveMember>
   <gml:curveMember>
    <gml:Curve gml:id="c10_2">
     <gml:segments>
      <gml:LineStringSegment>
       <gml:posList>8 3 9 3</gml:posList>
      </gml:LineStringSegment>
     </gml:segments>
    </gml:Curve>
   </gml:curveMember>  
  </gml:CompositeCurve>
 </ii:member>
<!-- <ii:member>
  <gml:CompositeCurve gml:id="object1">
   <gml:curveMember>
    <gml:Curve gml:id="c1">
     <gml:segments>
      <gml:LineStringSegment>
       <gml:posList>1 0 2 0 3 0 4 0 5 0</gml:posList>
      </gml:LineStringSegment>
      <gml:Arc>
       <gml:posList>4 0 4.5 1 5 0</gml:posList>
      </gml:Arc>
      <gml:LineStringSegment>
       <gml:posList>5 0 6 0 7 0</gml:posList>
      </gml:LineStringSegment>
     </gml:segments>
    </gml:Curve>
   </gml:curveMember>
   <gml:curveMember>
    <gml:Curve gml:id="c2">
     <gml:segments>
      <gml:LineStringSegment>
       <gml:posList>7 0 8 0</gml:posList>
      </gml:LineStringSegment>
     </gml:segments>
    </gml:Curve>
   </gml:curveMember>
   <gml:curveMember>
    <gml:Curve gml:id="c3">
     <gml:segments>
      <gml:LineStringSegment>
       <gml:posList>8 0 9 0 10 0</gml:posList>
      </gml:LineStringSegment>
     </gml:segments>
    </gml:Curve>
   </gml:curveMember>
  </gml:CompositeCurve>
 </ii:member>
  <ii:member>
   <gml:CompositeCurve gml:id="object2">
    <gml:curveMember>
     <gml:Curve gml:id="c4">
      <gml:segments>
       <gml:LineStringSegment>
        <gml:posList>1 0 2 0</gml:posList>
       </gml:LineStringSegment>
      </gml:segments>
     </gml:Curve>
    </gml:curveMember>
    <gml:curveMember>
     <gml:Curve gml:id="c5">
      <gml:segments>
       <gml:LineStringSegment>
        <gml:posList>2 0 3 0 4 0</gml:posList>
       </gml:LineStringSegment>
      </gml:segments>
     </gml:Curve>
    </gml:curveMember>
    <gml:curveMember>
     <gml:Curve gml:id="c6">
      <gml:segments>
       <gml:Arc>
        <gml:posList>4 0 4.5 1 5 0</gml:posList>
       </gml:Arc>
       <gml:LineStringSegment>
        <gml:posList>5 0 6 0 7 0</gml:posList>
       </gml:LineStringSegment>
      </gml:segments>
     </gml:Curve>
    </gml:curveMember>
    <gml:curveMember>
     <gml:Curve gml:id="c7">
      <gml:segments>
       <gml:LineStringSegment>
        <gml:posList>7 0 8 0 9 0 10 0</gml:posList>
       </gml:LineStringSegment>
      </gml:segments>
     </gml:Curve>
    </gml:curveMember>
   </gml:CompositeCurve>
  </ii:member>
 <ii:member>
  <gml:CompositeCurve gml:id="object3">
   <gml:curveMember>
    <gml:Curve gml:id="c8">
     <gml:segments>
      <gml:LineStringSegment>
       <gml:posList>1 0 2 0</gml:posList>
      </gml:LineStringSegment>
     </gml:segments>
    </gml:Curve>
   </gml:curveMember>
   <gml:curveMember>
    <gml:Curve gml:id="c9">
     <gml:segments>
      <gml:LineStringSegment>
       <gml:posList>2 0 3 0 4 0</gml:posList>
      </gml:LineStringSegment>
     </gml:segments>
    </gml:Curve>
   </gml:curveMember>
   <gml:curveMember>
    <gml:Curve gml:id="c10">
     <gml:segments>
      <gml:Arc>
       <gml:posList>4 0 4.25 1 5 0</gml:posList>
      </gml:Arc>
      <gml:LineStringSegment>
       <gml:posList>5 0 6 0 7 0</gml:posList>
      </gml:LineStringSegment>
     </gml:segments>
    </gml:Curve>
   </gml:curveMember>
   <gml:curveMember>
    <gml:Curve gml:id="c11">
     <gml:segments>
      <gml:LineStringSegment>
       <gml:posList>7 0 8 0 9 0 10 0</gml:posList>
      </gml:LineStringSegment>
     </gml:segments>
    </gml:Curve>
   </gml:curveMember>
  </gml:CompositeCurve>
 </ii:member>
 <ii:member>
  <gml:CompositeCurve gml:id="object4">
   <gml:curveMember>
    <gml:Curve gml:id="c12">
     <gml:segments>
      <gml:LineStringSegment>
       <gml:posList>1 0 2 0</gml:posList>
      </gml:LineStringSegment>
     </gml:segments>
    </gml:Curve>
   </gml:curveMember>
   <gml:curveMember>
    <gml:Curve gml:id="c13">
     <gml:segments>
      <gml:LineStringSegment>
       <gml:posList>2 0 4 0</gml:posList>
      </gml:LineStringSegment>
     </gml:segments>
    </gml:Curve>
   </gml:curveMember>
   <gml:curveMember>
    <gml:Curve gml:id="c14">
     <gml:segments>
      <gml:Arc>
       <gml:posList>4 0 4.5 1 5 0</gml:posList>
      </gml:Arc>
     </gml:segments>
    </gml:Curve>
   </gml:curveMember>
   <gml:curveMember>
    <gml:Curve gml:id="c15">
     <gml:segments>
      <gml:LineStringSegment>
       <gml:posList>5 0 7 0</gml:posList>
      </gml:LineStringSegment>
     </gml:segments>
    </gml:Curve>
   </gml:curveMember>
   <gml:curveMember>
    <gml:Curve gml:id="c16">
     <gml:segments>
      <gml:LineStringSegment>
       <gml:posList>7 0 10 0</gml:posList>
      </gml:LineStringSegment>
     </gml:segments>
    </gml:Curve>
   </gml:curveMember>
  </gml:CompositeCurve>
 </ii:member>-->
</ii:GeometryCollection>
