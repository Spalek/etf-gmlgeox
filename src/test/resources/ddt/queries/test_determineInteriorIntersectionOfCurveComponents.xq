import module namespace ggeo = 'https://modules.etf-validator.net/gmlgeox/2';

declare namespace gml = 'http://www.opengis.net/gml/3.2';
declare namespace ii = 'http://www.interactive-instruments.de/test';

declare variable $dbname external := 'GMLGEOX-JUNIT-TEST-DB-000';
let $members := db:open($dbname)//ii:member
let $init := ggeo:init($dbname)

let $geometries := $members/*
return
 <test_determineInteriorIntersectionOfCurveComponents>
  <test
   geom1="c1"
   geom2="c3">{exists(ggeo:determineInteriorIntersectionOfCurveComponents($geometries[@gml:id = 'c1'], $geometries[@gml:id = 'c3']))}</test>
  <test
   geom1="c1"
   geom2="c2">{exists(ggeo:determineInteriorIntersectionOfCurveComponents($geometries[@gml:id = 'c1'], $geometries[@gml:id = 'c2']))}</test>
  <test
   geom1="c2"
   geom2="c3">{exists(ggeo:determineInteriorIntersectionOfCurveComponents($geometries[@gml:id = 'c2'], $geometries[@gml:id = 'c3']))}</test>
  <test
   geom1="c4"
   geom2="p1">{exists(ggeo:determineInteriorIntersectionOfCurveComponents($geometries[@gml:id = 'c4'], $geometries[@gml:id = 'p1']))}</test>
     <test
   geom1="p2"
   geom2="p2">{exists(ggeo:determineInteriorIntersectionOfCurveComponents($geometries[@gml:id = 'p2'], $geometries[@gml:id = 'p2']))}</test>
  <test
   geom1="p2"
   geom2="c4">{exists(ggeo:determineInteriorIntersectionOfCurveComponents($geometries[@gml:id = 'p2'], $geometries[@gml:id = 'c4']))}</test>
  <test
   geom1="c5"
   geom2="c6">{exists(ggeo:determineInteriorIntersectionOfCurveComponents($geometries[@gml:id = 'c5'], $geometries[@gml:id = 'c6']))}</test>
  <test
   geom1="c7"
   geom2="c8">{exists(ggeo:determineInteriorIntersectionOfCurveComponents($geometries[@gml:id = 'c7'], $geometries[@gml:id = 'c8']))}</test>
  <test
   geom1="c9"
   geom2="c10">{exists(ggeo:determineInteriorIntersectionOfCurveComponents($geometries[@gml:id = 'c9'], $geometries[@gml:id = 'c10']))}</test>
 </test_determineInteriorIntersectionOfCurveComponents>
