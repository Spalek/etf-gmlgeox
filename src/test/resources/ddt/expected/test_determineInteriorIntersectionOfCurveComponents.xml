<test_determineInteriorIntersectionOfCurveComponents>
  <test geom1="c1" geom2="c3">true</test>
  <test geom1="c1" geom2="c2">false</test>
  <test geom1="c2" geom2="c3">true</test>
  <test geom1="c4" geom2="p1">true</test>
  <test geom1="p2" geom2="p2">false</test>
  <test geom1="p2" geom2="c4">false</test>
  <test geom1="c5" geom2="c6">false</test>
  <test geom1="c7" geom2="c8">false</test>
  <test geom1="c9" geom2="c10">true</test>
</test_determineInteriorIntersectionOfCurveComponents>